src/ConnectedMode.UnitTests/Suppressions/SuppressedIssueMatcherTests.cs

```diff
@@ -129,18 +129,53 @@ public void MatchExists_ResultDependsOnSuppressionState(string serverRuleId, int
             testSubject.SuppressionExists(issueToMatch).Should().Be(expectedResult);
         }
 
-        private IFilterableIssue CreateIssueToMatch(string ruleId, int? startLine, string lineHash)
+        [TestMethod]
+        // Module-level issues i.e. no file
+        [DataRow(null, null, true)]
+        [DataRow(null, "", true)]
+        [DataRow("", null, true)]
+        [DataRow("", "", true)]
+
+        // Module-level issues should not match non-module-level issues
+        [DataRow(@"any.txt", "", false)]
+        [DataRow(@"any.txt", null, false)]
+        [DataRow("", @"c:\any.txt", false)]
+        [DataRow(null, @"c:\any.txt", false)]
+        
+        // File issues
+        [DataRow(@"same.txt", @"c:\same.txt", true)]
+        [DataRow(@"SAME.TXT", @"c:\same.txt", true)]
+        [DataRow(@"same.TXT", @"c:\XXXsame.txt", false)]  // partial file name -> should not match
+        [DataRow(@"differentExt.123", @"a:\differentExt.999", false)] // different extension -> should not match
+        [DataRow(@"aaa\partial\file.cs", @"d:\partial\file.cs", false)]
+        // Only matching the local path tail, so the same server path can match multiple local files
```

Comment:
@duncanp-sonar the last 3 unit test cases here are what I mean by incorrect behavior. And I don't see how you can fix it without getting the solution folder.

Reply:
Edit: as we discussed offline, this was the case in the old code as well, and what the other SL flavors are doing as well. So let's stay with that.